src/psygnal/_signal.py

```diff
@@ -1037,29 +1033,20 @@ def __call__(
             asynchronous=asynchronous,
         )
 
+    def _flush_emit_queue(self) -> None:
+        if Signal.current_emitter() is not self:
```

Comment:
`Signal.current_emitter` is not thread safe, so depending on it in the emission mechanism may not be the best idea.

Reply:
that's fine, it can be stored on the object itself and incorporated into the `_lock`